lib/commands/favorite.go

```diff
@@ -11,15 +13,35 @@ import (
 	"github.com/urfave/cli/v2"
 )
 
-// ListFavorites prints out the users stored favorites. Extra information is
-// provided if the verbose flag is set.
+// ListFavorites prints out the users stored favorites and favorites from the Kion API.
+// Extra information is provided if the verbose flag is set.
 func (c *Cmd) ListFavorites(cCtx *cli.Context) error {
-	// map our favorites for ease of use
-	fNames, fMap := helper.MapFavs(c.config.Favorites)
+
+	// get the combined list of favorites from the CLI config and the Kion API (if compatible)
+	useApi := cCtx.App.Metadata["useFavoritesAPI"].(bool)
+	var apiFavorites []structs.Favorite
+	var err error
+	if useApi {
+		apiFavorites, _, err = kion.GetAPIFavorites(c.config.Kion.Url, c.config.Kion.ApiKey)
+		if err != nil {
+			fmt.Printf("Error retrieving favorites from API: %v\n", err)
+			return nil
```

Comment:
This should return the error instead of nil. The command will bubble it all the way up to exit.